anyway/parsers/location_extraction.py

```diff
@@ -278,57 +273,53 @@ def manual_filter_location_of_text(text):
                     )
 
     except Exception as _:
-        logging.info("could not filter text {0}".format(text))
-    if "כביש" in text:
-        filter_ind = min(filter_ind, text.find("כביש"))
-    if "שדרות" in text:
-        filter_ind = min(filter_ind, text.find("שדרות"))
-    if "רחוב" in text:
-        filter_ind = min(filter_ind, text.find("רחוב"))
-    if "מחלף" in text:
-        filter_ind = min(filter_ind, text.find("מחלף"))
-    if "צומת" in text:
-        filter_ind = min(filter_ind, text.find("צומת"))
-    if "סמוך ל" in text:
-        filter_ind = min(filter_ind, text.find("סמוך ל") + len("סמוך ל"))
-    if "ליד ה" in text:
-        filter_ind = min(filter_ind, text.find("ליד ה") + len("ליד ה"))
-    if "יישוב" in text:
-        filter_ind = min(filter_ind, text.find("יישוב"))
-    if "מושב" in text:
-        filter_ind = min(filter_ind, text.find("מושב"))
-    if "קיבוץ" in text:
-        filter_ind = min(filter_ind, text.find("קיבוץ"))
-    if "התנחלות" in text:
-        filter_ind = min(filter_ind, text.find("התנחלות"))
-    if "שכונת" in text:
-        filter_ind = min(filter_ind, text.find("שכונת"))
-    if "בדרך" in text:
-        filter_ind = min(filter_ind, text.find("בדרך"))
+        logging.exception("could not filter text {0}".format(text))
+
+    loc_tokens = [
+        "כביש",
+        "שדרות",
+        "רחוב",
+        "מחלף",
+        "צומת",
+        "יישוב",
+        "מושב",
+        "קיבוץ",
+        "התנחלות",
+        "שכונת",
+        "בדרך",
+    ]
+    for token in loc_tokens:
+        i = text.find(token)
+        if i >= 0:
+            filter_ind = min(filter_ind, i)
+
+    near_tokens = ["סמוך ל", "ליד ה"]  # maybe add: "ליד "
+    for token in near_tokens:
+        i = text.find(token)
+        if i >= 0:
+            filter_ind = min(filter_ind, i + len(token))
+
     if filter_ind != float("inf"):
         text = text[filter_ind:]
-    if "סמוך ל" in text:
-        text = (
-            text[(text.find("סמוך ל") + len("סמוך ל")) :] + "סמוך ל" + text[: text.find("סמוך ל")]
-        )
-    if "ליד ה" in text:
-        text = text[(text.find("ליד ה") + len("ליד ה")) :] + "ליד ה " + text[: text.find("ליד ה ")]
+
+    for token in near_tokens:
+        i = text.find(token)
+        if i >= 0:
+            text = text[i + len(token) :] + token + text[:i]
     return text
```

Comment:
The changes in this function have nothing to do with the PR. I can revert them if needed.

Reply:
They are good, maybe just include them in what the PR does.